wpilib/src/pneumatics.rs

```diff
@@ -5,210 +5,230 @@
+    /// Returns a bit mask.
+    pub fn all(&self) -> HalResult<i32> {
+        hal_call!(HAL_GetAllSolenoids(self.0))
     }
 
-    pub fn pcm_solenoid_voltage_sticky_fault_with_module(module: i32) -> bool {
-        maybe_hal_call!(HAL_GetPCMSolenoidVoltageStickyFault(module)).ok() != 0
+    pub fn solenoid_blacklist(&self) -> i32 {
+        maybe_hal_call!(HAL_GetPCMSolenoidBlackList(self.0)).ok()
     }
 
-    pub fn pcm_solenoid_voltage_sticky_fault(&self) -> bool {
-        Self::pcm_solenoid_voltage_sticky_fault_with_module(self.module)
+    pub fn solenoid_voltage_sticky_fault(&self) -> bool {
+        maybe_hal_call!(HAL_GetPCMSolenoidVoltageStickyFault(self.0)).ok() != 0
     }
 
-    pub fn pcm_solenoid_voltage_fault_with_module(module: i32) -> bool {
-        maybe_hal_call!(HAL_GetPCMSolenoidVoltageFault(module)).ok() != 0
+    pub fn solenoid_voltage_fault(&self) -> bool {
+        maybe_hal_call!(HAL_GetPCMSolenoidVoltageFault(self.0)).ok() != 0
     }
 
-    pub fn pcm_solenoid_voltage_fault(&self) -> bool {
-        Self::pcm_solenoid_voltage_fault_with_module(self.module)
+    pub fn clear_all_sticky_faults(&mut self) -> HalResult<()> {
+        hal_call!(HAL_ClearAllPCMStickyFaults(self.0))
     }
 
-    pub fn clear_all_pcm_sticky_faults_with_module(module: i32) {
-        maybe_hal_call!(HAL_ClearAllPCMStickyFaults(module)).ok();
+    pub fn solenoid(self, channel: i32) -> HalResult<Solenoid> {
+        Solenoid::with_module(self, channel)
     }
 
-    pub fn clear_all_pcm_sticky_faults(&self) {
-        Self::clear_all_pcm_sticky_faults_with_module(self.module);
+    pub fn double_solenoid(
+        self,
+        forward_channel: i32,
+        reverse_channel: i32,
+    ) -> HalResult<DoubleSolenoid> {
+        DoubleSolenoid::with_module(self, forward_channel, reverse_channel)
+    }
+
+    pub fn compressor(self) -> Compressor {
+        Compressor::with_module(self)
     }
 }
 
 #[derive(Debug)]
 pub struct Solenoid {
-    solenoid_handle: HAL_SolenoidHandle,
+    handle: HAL_SolenoidHandle,
     channel: i32,
-    module: SolenoidModule,
+    module: PneumaticsControlModule,
 }
 
 impl Solenoid {
-    /// Make a new solenoid with the given channel.
-    #[allow(clippy::new_ret_no_self)]
+    /// Create a new solenoid on the default PCM with the given channel.
     pub fn new(channel: i32) -> HalResult<Solenoid> {
-        Self::new_with_module(sensor_util::default_solenoid_module(), channel)
+        Self::with_module(PneumaticsControlModule::DEFAULT, channel)
     }
 
     /// If for some reason the Pneumatic Control Module is not on CAN module 0,
     /// you can use this constructor. Most people will never need this.
-    pub fn new_with_module(module_number: i32, channel: i32) -> HalResult<Solenoid> {
-        if !sensor_util::check_solenoid_module(module_number) {
-            return Err(HalError(0));
-        };
-
-        if !sensor_util::check_solenoid_channel(channel) {
```

Comment:
Do we not want to check if the channel is valid still?

Reply:
This is done in the HAL. Better to actually return an appropriate error that actually has an error message.